src/ellipse.rs

```diff
@@ -0,0 +1,225 @@
+use crate::{Angle, Point, Rect, Size, Vector};
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+use std::ops::{Add, AddAssign, Sub, SubAssign};
+
+#[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct Ellipse<T = f32> {
+    center: Point<T>,
+    radius: Size<T>,
+}
+
+impl<T: en::Num> Default for Ellipse<T> {
+    fn default() -> Self {
+        Self::unit()
+    }
+}
+
+impl<T: en::Num> Ellipse<T> {
+    pub fn new(center: Point<T>, radius: Size<T>) -> Self {
+        Self { center, radius }
+    }
+
+    pub fn unit() -> Self {
+        Self::new(Point::zero(), Size::square(T::one()))
+    }
+
+    pub fn zero() -> Self {
+        Self::from_radius(Size::zero())
+    }
+
+    pub fn from_radius(radius: Size<T>) -> Self {
+        Self::default().with_radius(radius)
+    }
+
+    pub fn from_center(center: Point<T>) -> Self {
+        Self::default().with_center(center)
+    }
+
+    pub fn with_radius(mut self, radius: Size<T>) -> Self {
+        self.radius = radius;
+        self
+    }
+
+    pub fn with_center(mut self, center: Point<T>) -> Self {
+        self.center = center;
+        self
+    }
+
+    pub fn add_radius(self, by: Size<T>) -> Self {
+        self.map_radius(move |radius: Size<T>| radius + by)
+    }
+
+    pub fn scale_radius(self, coeff: Vector<T>) -> Self {
+        self.map_radius(move |radius: Size<T>| radius.scale(coeff))
+    }
+
+    pub fn center(&self) -> Point<T> {
+        self.center
+    }
+
+    pub fn radius(&self) -> Size<T> {
+        self.radius
+    }
+
+    pub fn radius_squared(&self) -> Size<T> {
+        self.radius * self.radius.to_vector()
+    }
+
+    pub fn contains(&self, point: Point<T>) -> bool
+    where
+        T: float_cmp::ApproxEq,
```

Comment:
`en::Num` and `en::Float` are intended to be the alpha and omega of numeric traits, so it's generally verboten in gee to constrain on anything else. If this crate is essential, you'd need to incorporate it into the `en::Float` trait.

Reply:
Nahh, wasn't essential, just didn't realize there was an epsilon() in `en::Float` that I could use.